src/main/java/com/nutomic/syncthingandroid/fragments/DeviceSettingsFragment.java

```diff
@@ -142,6 +149,14 @@ else if (currentState != SyncthingService.State.ACTIVE) {
             }
             mDeviceId.setOnPreferenceClickListener(this);
         }
+
+        if (mDevice == null) {
+            // The device was removed from Syncthing and no reference was found
+            Log.d(TAG, "Empty mDevice");
+            getActivity().finish();
+            return;
+        }
```

Comment:
Can you put all of these in a single function that you call everywhere?

Also, it should be `Log.w`, and I'd just move the comment into the log (no need to have comment and log here).